app/controllers/api/v1/variant_details_controller.rb

```diff
@@ -0,0 +1,5 @@
+class Api::V1::VariantDetailsController < AuthenticatedApiController
+  def index
+    @variant_details = Visitor.find(params[:visitor_id]).assignments.includes(:variant_details).map(&:variant_detail)
```

Comment:
so, this was the result of a bit of back and forth about how best to optimize the queries for getting `VariantDetails`s for an assignment. an assignment has a split ID and a variant, which means it should be able to join directly to a `VariantDetail`, but I couldn't get ActiveRecord to do a compound join.

@jmileham any ideas?